EasyCommands.Tests/ScriptTests/ScriptTest.cs

```diff
@@ -191,48 +158,123 @@ public void Dispose()
         private void SetupMockBlocksByType<T>(string name, Mock<T>[] blockMocks) where T : class, IMyTerminalBlock
         {
             // Setup the mock blocks to mock their custom name to match the name provided
-            List<T> blockObjects = blockMocks.Select(b =>
-            {
-                b.Setup(x => x.CustomName).Returns(name);
-                return b.Object;
-            }).ToList();
+            foreach(Mock<T> block in blockMocks) {
+                block.Setup(x => x.CustomName).Returns(name);
+            }
+
             // Store the mock block objects by their type
-            mockedBlocksByName.Add(blockObjects);
+            SetupMockBlocksByType(blockMocks);
         }
 
         private void SetupMockBlocksByType<T>(Mock<T>[] blockMocks) where T: class, IMyTerminalBlock
         {
             // Store the mock block objects by their type
-            mockedBlocksByName.Add(blockMocks.Select(b => b.Object).ToList());
+            mockGrid.AddBlocks(blockMocks.Select(b => (IMyTerminalBlock)b.Object).ToList());
         }
 
-        public class GenericListDictionary
-        {
-            private Dictionary<Type, HashSet<Object>> dictionary = new Dictionary<Type, HashSet<object>>();
+        public class MockBlockGroup : IMyBlockGroup {
+            HashSet<IMyTerminalBlock> mockBlocks = new HashSet<IMyTerminalBlock>();
+            string name;
 
-            public void Add<T>(List<T> values)
-            {
-                Type key = typeof(T);
-                HashSet<object> valueObjects = values.Select(x => (Object)x).ToHashSet();
-                if (dictionary.ContainsKey(key))
-                {
-                    dictionary[key].UnionWith(valueObjects);
-                } else
-                {
-                    dictionary.Add(key, valueObjects);
-                }
-            }
-
-            public HashSet<T> GetValue<T>(Type type)
-            {
-                if (!dictionary.ContainsKey(type))
-                {
-                    return new HashSet<T>();
-                }
+            public MockBlockGroup(string name) {
+                this.name = name;
+            }
+
+            public string Name => name;
+
+            public void AddBlock(IMyTerminalBlock block) {
+                mockBlocks.Add(block);
+            }
+
+            public void AddBlocks(List<IMyTerminalBlock> blocks) {
+                blocks.ForEach(block => mockBlocks.Add(block));
+            }
```

Comment:
`mockBlocks.unionWith(blocks)` should do this.

Reply:
it doesn't quite.  i tried it and decided it's more characters and less efficient to do it.

What you actually have to do is:

`mockBlocks = new HashSet<IMyTerminalBlock>(blocks.Union(blocks));`

vs existing

`blocks.ForEach(block => mockBlocks.Add(block));`